narwhals/_pandas_like/dataframe.py

```diff
@@ -318,10 +318,8 @@ def with_columns(
     def rename(self, mapping: dict[str, str]) -> Self:
         return self._from_native_dataframe(self._native_dataframe.rename(columns=mapping))
 
-    def drop(self, *columns: str) -> Self:
-        return self._from_native_dataframe(
-            self._native_dataframe.drop(columns=list(columns))
-        )
+    def drop(self: Self, columns: str | list[str]) -> Self:
```

Comment:
does it need to be `str | list[str]`? didn't it get flattened one level above?

Reply:
I will check with mypy for internal use